src/shiv/bootstrap/environment.py

```diff
@@ -4,7 +4,7 @@
 """
 import json
 import os
-from typing import Any, Optional
+from typing import Any, Dict, Optional
```

Comment:
This is [deprecated](https://docs.python.org/3/library/typing.html#typing.Dict) since Python 3.9 and Python 3.8 is EOL (end-of-life) already.
I would not introduce an obsolete feature into a new code.

The `Mapping` is a better choice for generic types anyway (from `collections.abc.Mapping`).

Reply:
Falls into the same problem as `dict` on Python 3.8 unfortunately...
```py
Python 3.8.0 (default, Dec  9 2021, 17:53:27) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections.abc import Mapping
>>> from typing import Any
>>> hashes: Mapping[str, Any] = {}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'ABCMeta' object is not subscriptable
```

I see two courses of action here, either:
- Use my patch as is (`typing.Dict` is already used in [some parts of the code base](https://github.com/linkedin/shiv/blob/main/src/shiv/constants.py#L2) anyway)
- Use `dict` or `Mapping`, in which case you should also flag 1.0.7 onwards as being incompatible with Python 3.8

Reply:
Fair enough ...
The `typing.Mapping` alias to `collections.abc.Mapping` was also deprecated at the same time as `typing.Dict`.
So, it doesn't help for that.

Since there's [no scheduled release|https://docs.python.org/3/library/typing.html#deprecated-aliases] for the removal of the deprecated aliases, yet, I guess it's acceptable to continue using `Dict` — if 3.8 is going to be supported.

I'll leave it to @lorencarvalho to decide about this change.
If we decide to drop 3.8 support, it would be a good idea to increment at least the minor version of the package, though.